compiler+runtime/src/cpp/jank/read/parse.cpp

```diff
@@ -849,6 +849,37 @@ namespace jank::read::parse
       return object_source_info{ wrapped, start_token, str_end };
     }
 
+    if(sym->name == "inst")
+    {
+      auto str_result(next());
+
+      if(str_result.is_err())
+      {
+        return str_result;
+      }
+      else if(str_result.expect_ok().is_none())
+      {
+        return error::parse_invalid_reader_symbolic_value(
+          "The string literal after this '#inst' is missing.",
+          { start_token.start, latest_token.end });
+      }
+
+      auto const str_end(str_result.expect_ok().unwrap().end);
+
+
+      if(str_end.kind != lex::token_kind::string && str_end.kind != lex::token_kind::escaped_string)
+      {
+        return error::parse_invalid_reader_symbolic_value(
+          "The form after '#inst' must be a string literal.",
+          { start_token.start, latest_token.end });
+      }
+
+      auto const str(expect_object<obj::persistent_string>(str_result.expect_ok().unwrap().ptr));
+      auto const wrapped(make_box<obj::inst>(str->data));
```

Comment:
So, this will throw if the format is invalid, but that would just throw a normal exception with no source information. Thus, we won't get a nice compiler error output pointing at the string and saying the error. To do that, we'll need to catch that exception and then turn it into a compiler error.

We could get away with using `parse_invalid_reader_symbolic_value` again, but preferably we add a new `parse_invalid_inst` error kind.

Reply:
How would you feel about adding `parse_invalid_tagged_reader_value`? Then we can update `#cpp` and `#uuid` to this error instead?

Reply:
I would prefer that each of these reader tags has its own error. The reason is that these errors will relate to documentation, for search engine optimization. Each error id gets its own standalone web page, which should be the first search result for that page. In that page is a description of the error and common workarounds and fixes.

By using a super general error, we can't provide much useful info, or we'll have to provide way too much info, since the error could be related to `#cpp`, `#inst`, `#uuid`, etc. I think it's better to have a dedicated error, and thus a dedicated web page, for each one.

Reply:
Adding a new error involves:

1. Adding it to the enum in `error.hpp`
2. Adding it to the `kind_str` fn in `error.hpp`
3. Adding it to the `kind_to_message` in `error.cpp`
4. Adding a function with the same name to `error/analyze.hpp`
5. Implementing that function in `error/analyze.cpp` (they all just call `make_error`)

Reply:
Since this error is produced by the parser, should it be a parser error or should I include `#include <jank/error/analyze.hpp>` in the parser?

Reply:
`s/analyze/parse/g` in my last message.